File: Selenium/Activities/Activity20.py
# Import required packages from selenium
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/javascript-alerts")
    home_title = driver.title
    print("Title of the Home Page is : {}".format(home_title))

    # find and click the confirm button
    confirmButton = driver.find_element(By.ID, "prompt")
    confirmButton.click()

    # Wait till the alert shows up
    wait.until(expected_conditions.alert_is_present())

    # Switch focus to alert and print the alert message
    confirmAlert = driver.switch_to.alert
    print("Confirmation Alert Message : " + confirmAlert.text)

    time.sleep(2)
    confirmAlert.send_keys("Awesome")
    # Wait for 1 Second and click accept
    time.sleep(2)

    # Click Ok and close Alert
    confirmAlert.accept()

    # Wait for 3 Seconds and Close the Browser
    time.sleep(3)
    driver.close()

File: Selenium/Activities/Activity20.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity20 {
    public static void main(String[] args) throws InterruptedException {

        // Setup the Webdriver
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver= new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofMillis(1000));

        // Launch the URL and get the title
        driver.get("https://v1.training-support.net/selenium/javascript-alerts");
        String title = driver.getTitle();
        System.out.print("Title of the Page is : " + title + "\n");

        // find and click the confirm button
        WebElement confirmButton = driver.findElement(By.id("prompt"));
        confirmButton.click();

        // Wait till the alert shows up
        wait.until(ExpectedConditions.alertIsPresent());

        // Switch focus to alert and print the alert message
        Alert confirmAlert = driver.switchTo().alert();
        System.out.println("Confirmation Alert Message : " + confirmAlert.getText());

        Thread.sleep(2000);
        confirmAlert.sendKeys("Awesome");
        // Wait for 1 Second and click accept
        Thread.sleep(1000);

        // Click Ok and close Alert
        confirmAlert.accept();

        // Wait for 3 Seconds and close the browser
        Thread.sleep(3000);
        driver.close();

    }
}
